### PR TITLE
Implement JWT authentication for all API calls

### PR DESCRIPTION
**Pull request by kennym**

### CODE REVIEW

**Comment by kennym** [comment]
Fixes #47

**Comment by kennym** [comment]
Fixes #47

File: lib/stream/feed.rb [hunk 1]
```diff
@@ -118,12 +127,30 @@ def following(offset = 0, limit = 25, filter = [])
...
       uri = "/feed/#{@feed_url}/follows/#{target_feed_slug}:#{target_user_id}/"
-      @client.make_request(:delete, uri, @signature)
+      auth_token = create_jwt_token("follower", "delete")
+
+      @client.make_request(:delete, uri, auth_token, {auth_token: auth_token})
+    end
+
+    private
+
+    def create_jwt_token(resource, action, feed_id=nil, user_id=nil)
```
Comment (tbarbugli) [inline lib/stream/feed.rb]: we should make sure either  a value for `feed_id` or `user_id` is sent to this function

Comment (kennym) [inline lib/stream/feed.rb]: I'm setting the `feed_id` to be an instance variable if the `feed_id` is not provided: https://github.com/GetStream/stream-ruby/pull/50/files#diff-36085f1150dae50e5ff426a2f9b34811R148

**Comment by tbarbugli** [comment]
automatic merge does not work anymore (probably because of other PRs) can you have quick look at it?

**Comment by kennym** [comment]
@tbarbugli done

**Pull request merged**